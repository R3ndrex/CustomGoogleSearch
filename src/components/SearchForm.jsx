import { useState } from "react";
import { useNavigate } from "react-router-dom";

export default function SearchForm() {
    const [inputValue, setInputValue] = useState("");
    const navigate = useNavigate();
    function handleSubmit(e) {
        e.preventDefault();
        navigate(`${inputValue}/0`);
    }

    return (
        <form onSubmit={handleSubmit}>
            <strong>Gogle</strong>
            <input
                required
                type="search"
                id="search-input"
                name="search-input"
                placeholder="Enter search value"
                value={inputValue}
                onChange={(e) => setInputValue(e.currentTarget.value)}
            />
            <button type="submit">Search</button>
        </form>
    );
}
