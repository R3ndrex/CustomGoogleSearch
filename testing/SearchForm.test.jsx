import { describe, it, expect, vi } from "vitest";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SearchForm from "../src/components/SearchForm";

vi.mock("react-router-dom", () => ({
    ...vi.importActual("react-router-dom"),
    Link: vi.fn(),
}));

describe("SearchForm", () => {
    it("calls handleSubmit correctly", async () => {
        const handleSubmit = vi.fn();
        const user = userEvent.setup();
        render(
            <SearchForm handleSubmit={handleSubmit}>
                <input type="text" placeholder="search..." required />
                <button type="submit">Submit</button>
            </SearchForm>
        );

        const input = screen.getByPlaceholderText(/search.../i);

        await user.type(input, "hello kitty");
        await user.click(screen.getByText(/submit/i));

        expect(handleSubmit).toHaveBeenCalled();
    });
    it("doesnt call handleSubmit", async () => {
        const handleSubmit = vi.fn();
        const user = userEvent.setup();
        render(
            <SearchForm handleSubmit={handleSubmit}>
                <input type="text" placeholder="search..." required />
                <button type="submit">Submit</button>
            </SearchForm>
        );

        await user.click(screen.getByText(/submit/i));

        expect(handleSubmit).not.toHaveBeenCalled();
    });

    it("renders correctly", () => {
        const { container } = render(
            <SearchForm>
                <input type="text" placeholder="search..." />
                <button type="submit">Submit</button>
            </SearchForm>
        );
        expect(container).toMatchSnapshot();
    });
});
