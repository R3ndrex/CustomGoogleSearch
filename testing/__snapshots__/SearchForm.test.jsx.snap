// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchForm > renders correctly 1`] = `
<div>
  <nav>
    <form>
      <input
        placeholder="search..."
        type="text"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </nav>
</div>
`;
